pytorch_lightning/metrics/functional/hamming_loss.py

```diff
@@ -0,0 +1,68 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple, Union
+
+import torch
+from pytorch_lightning.metrics.classification.helpers import _input_format_classification
+
+
+def _hamming_loss_update(preds: torch.Tensor, target: torch.Tensor, threshold: float = 0.5) -> Tuple[torch.Tensor, int]:
+    preds, target, _ = _input_format_classification(preds, target, threshold=threshold)
+
+    correct = (preds == target).sum()
+    total = preds.numel()
+
+    return correct, total
+
+
+def _hamming_loss_compute(correct: torch.Tensor, total: Union[int, torch.Tensor]) -> torch.Tensor:
+    return 1 - correct.float() / total
+
+
+def hamming_loss(preds: torch.Tensor, target: torch.Tensor, threshold: float = 0.5) -> torch.Tensor:
```

Comment:
shall be loss separate from metrics?

Reply:
So this is called a loss only because better prediction result in lower values of the metric. It is not a loss in a sense that this is anything you could (or would want to) use to train your model. It's kind of like 1-accuracy.

Reply:
We could also name it Hamming distance, but I chose Hamming loss to stay in line with sklearn's naming

Reply:
@tadejsv I would name it distance to avoid confusion here...

Reply:
Alright, I'll rename it